src/geocodio/data.py

```diff
@@ -141,4 +141,66 @@ def formatted_addresses(self):
         """
         Returns a list of formatted addresses from the Location list
         """
-        return [item.formatted_address for item in self]
+        return [l.formatted_address for l in self]
+
+
+class LocationCollectionDict(dict):
+    """
+    A dict of Location objects, with dictionary lookup by address.
+    """
+
+    lookups = {}
+
+    def __init__(self, results_list, order="lat"):
+        """
+        Loads the individual responses into an internal list and uses the query
+        values as lookup keys.
+        """
+        results = {}
+        for key, result in results_list.items():
+            results[key]=Location(result["response"], order=order)
+            orig_query = result["query"]
+            lookup_key = json.dumps(orig_query) if isinstance(orig_query, dict) else orig_query
+            self.lookups[lookup_key] = key
+
+        super(LocationCollectionDict, self).__init__(results)
+        self.order = order
+
+    def get(self, key, default=KeyError):
```

Comment:
`LocationCollection` extends `list` and has no "expected behavior" for its `get` function, while `LocationCollectionDict` has the expected behavior of returning `default` if the key isn't there (and the default for `default` is `None`). Now, for parity and consistency, the `__getitem__` magic method can be overridden for both `LocationCollection` and `LocationCollectionDict` in order to have that be the use-case of throwing a KeyError if a key isn't present, otherwise returning the value (note that for `LocationCollection`, a numerical index would be a valid key in `__getitem__`). 

This way `LocationCollectionDict` maintains the expected behavior of a dict, and LocationCollection has some parity with `LocationCollectionDict` in how you can access elements using a key. I think this is the more intuitive approach for handling this. Other opinions are welcome.